### ISSUE TITLE
dart run build_runner build causes `Null check operator used on a null value'.

### ISSUE THREAD

**Issue by masahiro-ikemoto** [simolus3/moor#2175]
build_runner stops by `Null check operator used on a null value'.

Logs:
```
[SEVERE] drift_dev on lib/src/database.dart:

Null check operator used on a null value
package:drift_dev/src/writer/database_writer.dart 243:53        DatabaseWriter.createIndex
package:drift_dev/src/writer/database_writer.dart 114:17        DatabaseWriter.write
package:drift_dev/src/backends/build/drift_builder.dart 305:49  _DriftBuildRun._generateMonolithic
package:drift_dev/src/backends/build/drift_builder.dart 149:13  _DriftBuildRun.run
package:drift_dev/src/backends/build/drift_builder.dart 99:5    DriftBuilder.build
```

The source around the last of stack trace (database_writer.dart 243:53):
```
  static String createIndex(Scope scope, DriftIndex entity) {
    final sql = scope.sqlCode(entity.parsedStatement!);
    final index = scope.drift('Index');

    return '$index(${asDartLiteral(entity.schemaName)}, ${asDartLiteral(sql)})';
  }
```

I tryed to know what `entity' stops the process, and I've got:
- The entity.parsedStatement is null.
- The entity.createStmt is "CREATRE INDEX ...".

So I suspect it causes by "CREATE INDEX ..." statement in my xxx.drift that included by @DriftDatabase().

**Comment by masahiro-ikemoto** [comment]
For now, I tried the following, and build did success.
```
--- database_writer.dart.orig	2022-11-29 11:48:08.870259077 +0900
+++ database_writer.dart	2022-11-29 11:48:35.582461092 +0900
@@ -233,14 +233,14 @@
   }
 
   static String createTrigger(Scope scope, DriftTrigger entity) {
-    final sql = scope.sqlCode(entity.parsedStatement!);
+    final sql = entity.createStmt;
     final trigger = scope.drift('Trigger');
 
     return '$trigger(${asDartLiteral(sql)}, ${asDartLiteral(entity.schemaName)})';
   }
 
   static String createIndex(Scope scope, DriftIndex entity) {
-    final sql = scope.sqlCode(entity.parsedStatement!);
+    final sql = entity.createStmt;
     final index = scope.drift('Index');
 
     return '$index(${asDartLiteral(entity.schemaName)}, ${asDartLiteral(sql)})';
```

**Comment by simolus3** [comment]
Thanks for the report! This should be fixed in 0ee8a6a7f9f22541edaf679a50503c5b2c0e59f8, which I have also released as `drift_dev` version 2.3.2.
